projects/components/src/table/data/table-data-source.ts

```diff
@@ -1,8 +1,12 @@
 import { Observable } from 'rxjs';
 import { TableColumnConfig, TableFilter, TableSortDirection } from '../table-api';
 
-export interface TableDataSource<TResult, TCol extends TableColumnConfig = TableColumnConfig> {
-  getData(request: TableDataRequest<TCol>): Observable<TableDataResponse<TResult>>;
+export interface TableDataSource<TResult, TCol extends TableColumnConfig = TableColumnConfig, TDataTriggers = unknown> {
+  secondaryDataTriggers$?: Observable<TDataTriggers>; // TBD
+  getData(
+    request: TableDataRequest<TCol>,
+    secondaryDataTriggers?: TDataTriggers,
```

Comment:
we should be able to avoid this by making new request happen on a trigger change within the table. Is that something we can do? Getting the triggers as part of data source and then the consumer having to implement some logic dependent on them seems wierd.

We should not have to emit filters via these triggers again. These triggers can play the simple role of being signals while the getData method's `request` argument having the desired state always.

Reply:
> we should be able to avoid this by making new request happen on a trigger change within the table

This is exactly what we are doing. For requests triggered externally, we do need some `state` (e.g show inactive boolean). The purpose of this argument is to pass on that state to the getData method so that it can be used to build the correct gql query.